src/engine/pieces/queen.ts

```diff
@@ -1,13 +1,22 @@
 import Piece from './piece';
 import Player from '../player';
 import Board from '../board';
+import Square from '../square';
 
 export default class Queen extends Piece {
     public constructor(player: Player) {
         super(player);
     }
 
     public getAvailableMoves(board: Board) {
-        return new Array(0);
+        let availableMoves = new Array<Square>();
```

Comment:
```suggestion
        const availableMoves: Square[] = [];
```

Another thing is you would want to define this like the above as in typescript as 
- we don't need to instatiate arrays in typescript we just need to say it is of type Square
- arrays are all mutable in typescript so they can always be defined as const